### DISCUSSION TITLE
Enabling recaptcha with formr

### DISCUSSION THREAD

**Discussion by jhelvy** [rubenarslan/formr.org#518] [Q&A]
Has anyone used reCAPTCHA in a formr survey? I haven't looked too much into it, but I've gotten a lot of bots in prior surveys, so I'm trying to see if there's a way to integrate this at the start to weed out the bots.

**Comment by rubenarslan** [comment]
I think we'd probably have to integrate it, so that formr actually blocks progress if a bot is detected. You could just embed the Javascript and store the "quality" score, then exclude bots later. 

I'm a bit wary of using a Google solution in an otherwise tracking-free software. 

Have you tried using a dumb simple captcha? E.g. "Write 'orange' in this text box if you're not a bot."?
If you're not paying participants, I think this might be sufficient to weed out bots. If you're paying click workers, I'm not sure the captcha will help that much because they can still respond automatically once they've passed the captcha?

**Comment by jhelvy** [comment]
Good point about just keeping a measure of quality. I guess it doesn't really matter if the bots finish the survey as I can just drop them later, or I could screen them out (send them to a stop point) based on a quality threshold. Some of the dumb captchas don't really work as bots are smart enough to answer those questions now (LLMs are going to make this MUCH harder in the future).

Click workers are their own issue, and yes captchas won't stop them. We have to use other strategies. 

On a related point, have you seen a "hidden" question strategy used before on formr? I've heard of people making a simple multiple choice question but then hiding the form from human viewers (e.g. using CSS or something) so that the bots still answer it (they're just crawling the html). Makes it easy to ID the bots. I haven't tried this, but I was thinking about maybe using showif to test it. Do you think that might work? Like, just set the showif so that it never shows - is that how showif works? Would it still show up in the html?

**Comment by rubenarslan** [comment]
I have wondered whether hidden questions would work. You'd have to be careful in the implementation. I.e. if you hide the item with JS, bots using headless JS-enabled browsers might not be fooled. If you make it transparent, real people who navigate the survey by pressing tab might be mistaken for bots (and bots might actually skip these items too, if they've been told to). I guess it's all an arms race, so some strategies might work now but not later.

I do wonder who, other than clickworkers, writes a bot to fill out academic surveys? I had hoped the arms race wasn't as fierce as the arms race to get e.g. Gmail accounts, so we'd have to do less captcha-style stuff.

**Comment by jhelvy** [comment]
I've been thinking about building a ChatGPT-enabled bot to see what surveys I could get into on various platforms. Honestly I don't think it'd be all that difficult given the API and tools like langchain in python. I'm thinking of it sort of like pen testing these platforms. A lot of researchers rely on them, but with the LLM tools we now have there's no reason why we can't build a bot that acts just like a human and answers in human-like ways to surveys. Only reason I haven't tried this yet is because I don't want to get in trouble and / or break the law. Like, I can set it up to return the money, etc., but my bot responses will still be in peoples' surveys. I guess I could notify them immediately after taking the survey, etc. Just seems like someone should do this to show that it can be done.
